src/components/TodoItem.tsx

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable */
+import React, { useState, Fragment, useRef } from 'react';
+import { Button } from './Button';
+import { Todo } from '../types/Todo';
+import { editTodo } from '../api/todos';
+import { ErroMessage } from '../utils/errorMessages';
+
+type Props = {
+  todo: Todo;
+  isLoading: number[];
+  onDelete: (todoItem: Todo, shouldFocus: boolean) => void;
+  addEditedTodos?: (editedTodo: Todo) => void;
+  handleErrorMessages?: (newErrorMessage: ErroMessage) => void;
+  setIsLoading?: React.Dispatch<React.SetStateAction<number[]>>;
+};
```

Comment:
please - refactor the props to pass only the information you need:

```suggestion
type Props = {
  todo: Todo;
  isLoading: boolean;
  onDelete: (todoItem: Todo, shouldFocus: boolean) => void;
  addEditedTodo?: (editedTodo: Todo) => void;
  handleErrorMessages?: (newErrorMessage: ErroMessage) => void;
  setIsLoading?: (value: boolean) => void
};
```

with this you will improve the code structure and reduce the number of the possible errors

Reply:
Thx ) i fix everything  except this one  "setIsLoading?: React.Dispatch<React.SetStateAction<number[]>>" Its because i cant pass tests without it for some reason...)